Telerik.Sitefinity.Frontend.EmailCampaigns/Mvc/Models/UnsubscribeForm/UnsubscribeFormModel.cs

```diff
@@ -115,25 +115,32 @@ public UnsubscribeFormViewModel CreateViewModel()
         }
 
         /// <inheritDoc/>
-        public virtual void ExecuteAction(string subscriberId, string issueId, bool shouldSubscribe)
+        public virtual void ExecuteAction(string subscriberId, string issueId, string listId, bool shouldSubscribe)
         {
             var newslettersManager = NewslettersManager.GetManager(this.ProviderName);
 
             var issueGuid = Guid.Empty;
             var subscriberGuid = Guid.Empty;
+            var mailingListGuid = Guid.Empty;
 
-            if (!string.IsNullOrEmpty(subscriberId) && Guid.TryParse(subscriberId, out subscriberGuid)
-                && (!string.IsNullOrEmpty(issueId) && Guid.TryParse(issueId, out issueGuid)))
+            if (!string.IsNullOrEmpty(subscriberId) && Guid.TryParse(subscriberId, out subscriberGuid))
             {
                 Guid mailingListId = Guid.Empty;
-                var issue = newslettersManager.GetIssue(issueGuid);
-                if (issue != null)
+                if (!string.IsNullOrEmpty(listId) && Guid.TryParse(listId, out mailingListGuid) && mailingListGuid != Guid.Empty)
```

Comment:
IN both cases you could just do a Guid.TryParse. It will handle the IsNullOrEmpty clause